### PR TITLE
adds monthly spending chart, no spending list items yet

### PR DESCRIPTION
**Pull request by bamatson**

### CODE REVIEW

File: app/src/main/java/io/mattcarroll/androidtesting/spending/ChartSectionViewModel.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    private final BigDecimal cashAmount;
+    private final String category;
+
+    public ChartSectionViewModel(@NonNull BigDecimal cashAmount, @NonNull String category) {
+        this.cashAmount = cashAmount;
+        this.category = category;
+    }
+
+    @NonNull
+    BigDecimal getCashAmount() {
```
Comment (matthew-carroll) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/ChartSectionViewModel.java]: I think you got the exposure backwards in this class.  The constructor should match the package private of the class, but the methods should all be public.

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    private MonthlySpendingChart monthlySpendingChart;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_monthly_spending);
+        init();
+    }
+
+    private void init() {
```
Comment (matthew-carroll) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java]: Breaking this out seems superfluous...

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        monthlySpendingChart = (MonthlySpendingChart) findViewById(R.id.monthly_spending_chart);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        MonthlySpendingChartPresenter presenter = new MonthlySpendingChartPresenter();
```
Comment (matthew-carroll) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java]: 2 issues here:
1. I doubt you want to construct this every onResume().  It probably belongs in onCreate().
2. You probably don't want to declare it locally.  If you're gonna change that later then it's fine, but I wouldn't leave it that way.

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        monthlySpendingChart.setViewModels(presenter.presentChart());
```
Comment (matthew-carroll) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java]: We should unify on this naming.  If you want to go with present[Stuff]() then we'll need to change the presenters I created to do the same.

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChart.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+    public MonthlySpendingChart(@NonNull Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+        init();
+    }
+
+    private void init() {
+        LayoutInflater.from(getContext()).inflate(R.layout.monthly_spending_chart, this, true);
+        pieChart = (PieChart) findViewById(R.id.piechart);
+
```
Comment (matthew-carroll) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChart.java]: any reason for this blank line?

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChart.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        List<PieEntry> entries = new ArrayList<>(viewModels.size());
+
+        for (ChartSectionViewModel viewModel : viewModels) {
+            PieEntry entry = toPieEntry(viewModel);
+            entries.add(entry);
+        }
+
+        PieDataSet dataSet = new PieDataSet(entries, "Monthly Spending Breakdown");
+
+        dataSet.setSliceSpace(3f);
```
Comment (matthew-carroll) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChart.java]: magic numbers?

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChartPresenter.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import android.support.annotation.NonNull;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.List;
+
+class MonthlySpendingChartPresenter {
+
+    @NonNull
+    List<ChartSectionViewModel> presentChart() {
```
Comment (matthew-carroll) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChartPresenter.java]: Methods should be public.

File: app/src/main/res/layout/monthly_spending_chart.xml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<merge xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (matthew-carroll) [inline app/src/main/res/layout/monthly_spending_chart.xml]: xmlns should go on the next line to match the convention so far

File: app/src/main/res/layout/monthly_spending_chart.xml [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+       android:layout_width="wrap_content"
+       android:layout_height="wrap_content">
```
Comment (matthew-carroll) [inline app/src/main/res/layout/monthly_spending_chart.xml]: caret should go on the next line

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java (hunk 4 above)
Comment (bamatson) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java]: Not sure which presenters you're talking about. Feel free to change this in a follow up PR.

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChart.java (hunk 5 above)
Comment (bamatson) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChart.java]: nope

File: app/src/main/java/io/mattcarroll/androidtesting/spending/ChartSectionViewModel.java (hunk 1 above)
Comment (bamatson) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/ChartSectionViewModel.java]: done

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java (hunk 2 above)
Comment (bamatson) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java]: fixed

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java (hunk 3 above)
Comment (bamatson) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingActivity.java]: done and done

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChart.java (hunk 6 above)
Comment (bamatson) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChart.java]: fixed. the styling is arbitrary anyway.

File: app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChartPresenter.java (hunk 7 above)
Comment (bamatson) [inline app/src/main/java/io/mattcarroll/androidtesting/spending/MonthlySpendingChartPresenter.java]: done

File: app/src/main/res/layout/monthly_spending_chart.xml (hunk 8 above)
Comment (bamatson) [inline app/src/main/res/layout/monthly_spending_chart.xml]: done

File: app/src/main/res/layout/monthly_spending_chart.xml (hunk 9 above)
Comment (bamatson) [inline app/src/main/res/layout/monthly_spending_chart.xml]: done

**Pull request merged**